wdl-doc/src/lib.rs

```diff
@@ -67,282 +170,198 @@ impl Document {
-    analyzer.add_directory(abs_path.clone()).await?;
+    analyzer.add_directory(workspace_abs_path.clone()).await?;
     let results = analyzer.analyze(()).await?;
 
+    let mut docs_tree = if let Some(ss) = stylesheet {
+        docs_tree::DocsTree::new_with_stylesheet(docs_dir.clone(), ss)?
+    } else {
+        docs_tree::DocsTree::new(docs_dir.clone())
+    };
+
     for result in results {
-        let cur_path = result
-            .document()
-            .uri()
-            .to_file_path()
-            .expect("URI should have a file path");
-        let relative_path = match cur_path.strip_prefix(&abs_path) {
-            Ok(path) => path,
-            Err(_) => &PathBuf::from("external").join(cur_path.strip_prefix("/").unwrap()),
+        let uri = result.document().uri();
+        let rel_wdl_path = match uri.to_file_path() {
+            Ok(path) => match path.strip_prefix(&workspace_abs_path) {
+                Ok(path) => path.to_path_buf(),
+                Err(_) => {
+                    PathBuf::from("external").join(path.components().skip(1).collect::<PathBuf>())
+                }
+            },
+            Err(_) => PathBuf::from("external").join(
+                uri.path()
+                    .strip_prefix("/")
+                    .expect("URI path should start with /"),
+            ),
         };
-        let cur_dir = docs_dir.join(relative_path.with_extension(""));
+        let cur_dir = docs_dir.join(rel_wdl_path.with_extension(""));
         if !cur_dir.exists() {
             std::fs::create_dir_all(&cur_dir)?;
         }
-        let name = cur_dir
-            .file_name()
-            .expect("current directory should have a file name")
-            .to_string_lossy();
         let ast_doc = result.document().node();
         let version = ast_doc
             .version_statement()
-            .expect("document should have a version statement")
-            .version();
-        let preamble = fetch_preamble_comments(ast_doc.clone());
+            .expect("Document should have a version statement");
         let ast = ast_doc.ast().unwrap_v1();
 
-        let document = Document::new(name.to_string(), version, preamble);
-
-        let index = cur_dir.join("index.html");
-        let mut index = tokio::fs::File::create(index).await?;
-
-        index.write_all(document.to_string().as_bytes()).await?;
+        let mut local_pages = Vec::new();
 
         for item in ast.items() {
             match item {
                 DocumentItem::Struct(s) => {
-                    let struct_name = s.name().as_str().to_owned();
-                    let struct_file = cur_dir.join(format!("{}-struct.html", struct_name));
-                    let mut struct_file = tokio::fs::File::create(struct_file).await?;
+                    let name = s.name().as_str().to_owned();
+                    let path = cur_dir.join(format!("{}-struct.html", name));
```

Comment:
It might be a minor thing, but I would advocate for `struct-{}.html`, since that more closely follows the hierarchy here.

Reply:
Okay that's fine I suppose. I definitely think that classifying it into buckets of struct/workflow/task is a higher-order ranking of the objects. But it would probably make more sense to do `/struct/<name`, `/task/<name>`, and `/workflow/<name>` in that case.

Reply:
Somewhat related, I can see some repositories wanting the sidebar to look like:

- structs
-- struct 1
-- struct 2
- tasks
-- task 1
-- task 2

Rather than being grouped by source file. Clay's naming convention would make that easy to implement.